# Exercise 11

# Question 1: redo HW03 using python build-in functions.

In EX07, we have worked together to write a code to read the file earthquake.csv and assign each column to a list. We now want to further examine this earthquake catalog.

The code is shown below

In [4]:
import numpy as np
fp=open("./data/earthquake.csv","r")
lines=fp.readlines()
fp.close()
year=[];month=[];day=[];time=[];mag=[];lon=[];lat=[];depth=[];region=[]
for line in lines[1:]:
    if "\"" in line:
        temp=line.split(",")
        year.append(temp[0]);month.append(temp[1])
        day.append(temp[2]);time.append(temp[3])
        mag.append(temp[4]);lon.append(temp[5]);lat.append(temp[6])
        depth.append(temp[7]);region.append(temp[8]+","+temp[9])
    else:
        temp=line.split(",")
        year.append(temp[0]);month.append(temp[1])
        day.append(temp[2]);time.append(temp[3])
        mag.append(temp[4]);lon.append(temp[5]);lat.append(temp[6])
        depth.append(temp[7]);region.append(temp[8])

The code above will give you variables of 'year', 'month', 'day', 'time', 'mag', 'lon', 'lat', 'depth', and 'regoin'. All these variables have `list` data type.
__Be careful that  Each element within the lists has `string` data type.__ Therefore, you will need to first convert strings to numbers before apply mathmatic operations on it.

In [5]:
# To convert a string to a number, we use float() or int() functions
a="10"
b=float(a)
c=int(a)
print(a,b,c)

10 10.0 10


Convert all elements in the lists of 'mag','lon','lat',and 'depth' to floating numbers, and all elements in 'year', 'month', and 'day' to integer numbers. Remember, only change the data type of each element, and do not change the variable names.

In [8]:
for i in range(0,len(mag)):
    mag[i]=float(mag[i])
    lon[i]=float(lon[i])
    lat[i]=float(lat[i])
    depth[i]=float(depth[i])
    year[i]=int(year[i])
    month[i]=int(month[i])
    day[i]=int(day[i])

(a) Use python's build-in function to calculate the average magnitude of the earthquakes in this catalog.
Hint: you need to convert list to numpy array first because the mean function is in numpy array, not in list.

In [7]:
np.array(mag).mean()

1.783

I get: 1.783

(b) Find the maximum depth of the earthquakes

I get: 573.9

(c) In which region(s) does the largest earthquake(s) happen?
Hint: use the 'index' function in lists

I get: "MINDANAO, PHILIPPINES"

(d) In which region(s) and at what year, month, day and time do the deepest earthquake(s) happen?
- Hint: use the 'index' function in lists

I get: FLORES SEA 2019 2 10 22:54:37

## Question 2

- Make a file named 'fun.py' in the current folder
- Write a function named 'sinx' in the file 'fun.py' to compute $sin(x)/x$
- import the file
- calculate sin(2)/2 by calling 'fun.sinx(2)'
- Hint: when using sin(x), you need to import 'numpy' first